tests/unit/peapods/pods/test_pods.py

```diff
@@ -192,24 +193,30 @@ def foo(self, docs: DocumentArray, **kwargs):
 
 
 @pytest.mark.slow
-@pytest.mark.asyncio
-async def test_pod_rolling_update():
-    args_list = ['--replicas', '3']
+@pytest.mark.parametrize('shards', [1, 2])
+def test_pod_rolling_update(shards):
+    args_list = ['--replicas', '7']
+    args_list.extend(['--shards', str(shards)])
     args = set_pod_parser().parse_args(args_list)
     args.uses = 'AppendParamExecutor'
     args.uses_with = {'param': 10}
     with Pod(args) as pod:
-        response_texts = await _send_requests(pod)
-        assert 2 == len(response_texts)
-        assert all(text in response_texts for text in ['10', 'client'])
-
-        print('do rolling update')
-        await pod.rolling_update(uses_with={'param': 20})
-        response_texts = await _send_requests(pod)
-        print(response_texts)
-        assert 2 == len(response_texts)
-        assert all(text in response_texts for text in ['20', 'client'])
-        assert '10' not in response_texts
+
+        async def run_async_test():
+            response_texts = await _send_requests(pod)
+            assert 2 == len(response_texts)
+            assert all(text in response_texts for text in ['10', 'client'])
+
+            await pod.rolling_update(uses_with={'param': 20})
+            response_texts = await _send_requests(pod)
+            assert 2 == len(response_texts)
+            assert all(text in response_texts for text in ['20', 'client'])
+            assert '10' not in response_texts
+
+        p = Process(target=run_async_test)
```

Comment:
isn't it weird to call a function defined in a different context? I do not understand

Reply:
I can slightly improve the structure here, but I need to run the test code (I think at least the request sending) in a different process, otherwise we have multiple grpc clients running concurrently in the same process which messes up things